File: Solution/CPP/Sahil/Quiz/src/Question.cpp
#include "../include/question.h"
#include "../include/QuestionBank.h"
#include <string>
#include <iostream>
#include <algorithm>
#include <vector>


Question::Question() {
    // Default constructor
}
int Question::getId() {
    return id;
}   
void Question::setId(int id) {
    this->id = id;
}
string Question::getSubject() {
    return subject;
}
void Question::setSubject(string subject) {
    this->subject = subject;
}
string Question::getTitle() {
    return title;
}
void Question::setTitle(string title) {
    this->title = title;
}

string Question::getOptionA() {
    return option_a;
}
void Question::setOptionA(string option_a) {
    this->option_a = option_a;
}
string Question::getOptionB() {
    return option_b;
}
void Question::setOptionB(string option_b) {
    this->option_b = option_b;
}
string Question::getOptionC() {
    return option_c;
}
void Question::setOptionC(string option_c) {
    this->option_c = option_c;
}
string Question::getOptionD() {
    return option_d;
}
void Question::setOptionD(string option_d) {
    this->option_d = option_d;
}
char Question::getAnswerKey() {
    return answer_key;
}
void Question::setAnswerKey(char answer_key) {
    this->answer_key = answer_key;
}
int Question::getEvaluationId() {
    return evaluation_id;
}
void Question::setEvaluationId(int evaluation_id) {
    this->evaluation_id = evaluation_id;
}

void Question::display() {
    cout << "ID: " << id << endl;
    cout << "Subject: " << subject << endl;
    cout << "Title: " << title << endl;
    cout << "Option A: " << option_a << endl;
    cout << "Option B: " << option_b << endl;
    cout << "Option C: " << option_c << endl;
    cout << "Option D: " << option_d << endl;
    cout << "Answer Key: " << answer_key << endl;
    cout << "Evaluation ID: " << evaluation_id << endl;
}
Question::~Question() {
    // Destructor
    // No dynamic memory to clean up, but can be overridden in derived classes if needed
}

File: Solution/Java/NikitaB/TFLAssesment/src/main/java/com/tap/assesment/Entity/Question.java
package com.tap.assesment.Entity;

import java.io.*;

public class Question implements Serializable {
    private int id;
    private String title;
    private String optionA;
    private String optionB;
    private String optionC;
    private String optionD;
    private char correctAnswer;
    private String subject;
    private int evaluationCriteria;

    public Question(){
        id=0;
        title=null;
        optionA=null;
        optionB=null;
        optionC=null;
        optionD=null;
        correctAnswer='\0';
        subject=null;
        evaluationCriteria=28;
    }

    public Question(int id,String title,String optionA,String optionB,String optionC,String optionD,char correctAnswer,String subject,int evaluationCriteria){
        this.id=id;
        this.title=title;
        this.optionA=optionA;
        this.optionB=optionB;
        this.optionC=optionC;
        this.optionD=optionD;
        this.correctAnswer=correctAnswer;
        this.subject=subject;
        this.evaluationCriteria=evaluationCriteria;
    }

        public void setId(int id){this.id=id;}
        public int getId(){return id;}

        public void setTitle(String title){this.title=title;}
        public String getTitle(){return title;}
    
        public void setOptionA(String a){this.optionA=a;}
        public String getOptionA(){return optionA;}

        public void setOptionB(String b){this.optionB=b;}
        public String getOptionB(){return optionB;}

        public void setOptionC(String c){this.optionC=c;}
        public String getOptionC(){return optionC;}

        public void setOptionD(String d){this.optionD=d; }
        public String getOptionD(){return optionD;}

        public void setCorrectAnswer(char c){this.correctAnswer=c;}
        public char getCorrectAnswer(){return correctAnswer;}

        public void setSubject(String subject){this.subject=subject;}
        public String getSubject(){return subject;}

        public void setEvaluationCriteria(int criteria){this.evaluationCriteria=criteria;}
        public int getEvaluationCriteria(){return evaluationCriteria;}

        public String toString(){
            return ("Id : "+id+" "+"Title : "+title+" "+"Option A : "+optionA+" "+"Option B : "+optionB+" "+"Option C : "+optionC+"Option D : "+optionD+"Correct Answer : "+correctAnswer+"Evaluation Criteria"+evaluationCriteria);
        }
}
